src/Microsoft.TemplateEngine.TemplateLocalizer.Core/KeyCreators/ChildValueKeyCreator.cs

```diff
@@ -28,12 +28,12 @@ public string CreateKey(JsonElement element, string? elementName, string? parent
         {
             if (!element.TryGetProperty(MemberPropertyName, out JsonElement keyProperty) || keyProperty.ValueKind != JsonValueKind.String)
             {
-                string owningElementName = (parentElementName == null ? elementName : (parentElementName + "." + elementName)) ?? string.Empty;
+                string owningElementName = (parentElementName == null ? elementName : (parentElementName + TemplateStringExtractor._keySeparator + elementName)) ?? string.Empty;
```

Comment:
should we add the logic to TE forbidding using this symbol in name? (might be different PR)

Reply:
Good idea. Created an issue here https://github.com/dotnet/templating/issues/3127.